application/controllers/Kalkun.php

```diff
@@ -288,10 +288,18 @@ function settings($type = NULL)
 		{
 			$option = $this->input->post('option');
 			// check password
-			if ($option === 'password' && ! password_verify($this->input->post('current_password'), $this->Kalkun_model->get_setting()->row('password')))
+			if ($option === 'password')
 			{
-				$this->session->set_flashdata('notif', tr_raw('Wrong password'));
-				redirect('settings/'.$option);
+				if ($this->config->item('demo_mode') && intval($this->session->userdata('id_user')) === 1)
```

Comment:
`intval($this->session->userdata('id_user'))` is this check required?

Reply:
I prefer to do it, I don't know if ci's sessions is from cookie or wherever. Doesn't harm IMHO